### PR TITLE
refactor (jkube-kit/common) : Migrate all tests from JMockit to Mockito (#1545)

### PR DESCRIPTION
**Pull request by baruKreddy**
Signed-off-by: baruKreddy [bhargavirkotha@gmail.com](mailto:bhargavirkotha@gmail.com)

Fixes https://github.com/eclipse/jkube/issues/1545

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] My code follows the style guidelines of this project
 - [ ] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsTest.java [hunk 1]
```diff
@@ -114,37 +116,41 @@ public void getStringValueTest() {
   @Test
   public void getFromSystemPropertyWithPropertiesAsFallbackHasKeyInSystemShouldReturnSystemValue() {
     // Given
-    new SystemMock().put("key", "systemValue");
+    //new SystemMock().put("key", "systemValue");
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsTest.java:119]:
```suggestion

```

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsTest.java [hunk 2]
```diff
@@ -114,37 +116,41 @@ public void getStringValueTest() {
...
     // Then
     assertThat(result).isEqualTo("fallbackValue");
+    System.clearProperty("not-the-key");
   }
 
   @Test
   public void getFromSystemPropertyWithPropertiesAsFallbackHasNotKeyShouldReturnNull() {
     // Given
-    new SystemMock().put("not-the-key", "systemValue");
+    System.setProperty("not-the-key", "systemValue");
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsTest.java:146]: https://github.com/eclipse/jkube/pull/1609/files#r897746602

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java [hunk 3]
```diff
@@ -397,7 +399,7 @@ public void testLoadTimestampShouldLoadFromFile() throws Exception {
     @Test
     public  void testIsWindowsFalse(){
         //Given
-        new SystemMock().put("os.name", "random");
+        System.setProperty("os.name", "random");
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java:402]: https://github.com/eclipse/jkube/pull/1609/files#r897746602

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java [hunk 4]
```diff
@@ -407,7 +409,7 @@ public  void testIsWindowsFalse(){
     @Test
     public  void testIsWindows(){
         //Given
-        new SystemMock().put("os.name", "windows");
+        System.setProperty("os.name", "windows");
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java:412]: https://github.com/eclipse/jkube/pull/1609/files#r897746602

Comment (rohanKanojia) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java:412]: after setting system property, you need to clear it as well

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java (hunk 3 above)
Comment (rohanKanojia) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java:402]: Also clear this system property at the end of test

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/JKubeProjectUtilTest.java [hunk 5]
```diff
@@ -177,7 +185,7 @@ public void getFinalOutputArtifact_withArtifactAndBuildFinalNameAndPackaging_ret
   @Test
   public void getProperty_whenSystemPropertyPresent_returnsSystemProperty() {
     // Given
-    new SystemMock().put("jkube.testProperty", "true");
+    System.setProperty("jkube.testProperty", "true");
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/JKubeProjectUtilTest.java:188]: after setting system property, you need to clear it as well

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KubernetesHelperTest.java [hunk 6]
```diff
@@ -24,27 +24,13 @@
 import java.util.Map;
 import java.util.Set;
 
-import io.fabric8.kubernetes.api.model.GenericKubernetesResource;
-import io.fabric8.kubernetes.api.model.GenericKubernetesResourceBuilder;
-import io.fabric8.kubernetes.api.model.HTTPHeader;
-import io.fabric8.kubernetes.api.model.KubernetesList;
-import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
-import io.fabric8.kubernetes.api.model.LabelSelector;
+import io.fabric8.kubernetes.api.model.*;
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KubernetesHelperTest.java:27]: Please don't use Wildcard imports.

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/OpenshiftHelperTest.java [hunk 7]
```diff
@@ -36,60 +36,64 @@
...
+import static org.mockito.Mockito.when;
+@RunWith(MockitoJUnitRunner.class)
 public class OpenshiftHelperTest {
 
+    @Mock
+    KubernetesClient kc;
+
+    @Mock
+    OpenShiftClient oc;
+/*
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/OpenshiftHelperTest.java:48]: Why have you commented this test?

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/SpringBootUtilTest.java [hunk 8]
```diff
@@ -13,14 +13,18 @@
...
 package org.eclipse.jkube.kit.common.util;
 
-import mockit.Expectations;
-import mockit.Mocked;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.apache.commons.io.FileUtils;
 import org.eclipse.jkube.kit.common.Dependency;
 import org.eclipse.jkube.kit.common.JavaProject;
+import org.junit.Before;
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/SpringBootUtilTest.java:21]: unused import

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsTest.java [hunk 9]
```diff
@@ -19,11 +19,13 @@
...
 import org.apache.commons.lang3.RandomStringUtils;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.fail;
-
+@RunWith(MockitoJUnitRunner.class)
```
Comment (manusa) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/ConfigsTest.java:28]: This is not needed unless you use Mockito specific annotations.

In general we prefer not to use specific runners (this would also ease the transition to JUnit5)

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java [hunk 10]
```diff
@@ -397,21 +399,25 @@ public void testLoadTimestampShouldLoadFromFile() throws Exception {
...
+        // TODO : Replace this when https://github.com/eclipse/jkube/issues/958 gets fixed
         //When
         boolean result= EnvUtil.isWindows();
         //Then
         assertThat(result).isTrue();
+        System.clearProperty("os.name");
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java:420]: This is an already existing system property. We need to be careful while clearing it out. Let's change it like this:

```java
    @Test
    public  void testIsWindows(){
        String oldOsName = System.getProperty("os.name");
        System.setProperty("os.name", "windows");

        //Given


        //When

        //Then

        System.setProperty("os.name", oldOsName);
    }
```

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/JKubeProjectUtilTest.java [hunk 11]
```diff
@@ -14,28 +14,40 @@
...
 
 import java.io.File;
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.Properties;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
 
 @SuppressWarnings("ResultOfMethodCallIgnored")
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/JKubeProjectUtilTest.java:41]: Remove this redundant suppression

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/JKubeProjectUtilTest.java [hunk 12]
```diff
@@ -14,28 +14,40 @@
...
 import org.eclipse.jkube.kit.common.Dependency;
 import org.eclipse.jkube.kit.common.JavaProject;
 import org.eclipse.jkube.kit.common.SystemMock;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Incubating;
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/JKubeProjectUtilTest.java:27]: Please remove all unused imports here

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KubernetesHelperTest.java [hunk 13]
```diff
@@ -53,23 +54,30 @@
...
+import org.mockito.junit.MockitoJUnitRunner;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
-
+import static org.mockito.Mockito.*;
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KubernetesHelperTest.java:69]: Please don't use [wildcard imports](https://stackoverflow.com/a/3348855/2830116).

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KubernetesHelperTest.java [hunk 14]
```diff
@@ -53,23 +54,30 @@
...
+@RunWith(MockitoJUnitRunner.class)
 public class KubernetesHelperTest {
 
     private KitLogger logger;
 
+    @Mock
+    KubernetesClient kubernetesClient;
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KubernetesHelperTest.java:76]:
```suggestion
    private KubernetesClient kubernetesClient;
```

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KubernetesHelperTest.java [hunk 15]
```diff
@@ -53,23 +54,30 @@
...
+
+    @Mock
+    Resource<Service> svcResource;
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KubernetesHelperTest.java:79]:
```suggestion
    private Resource<Service> svcResource;
```

**Comment by rohanKanojia** [comment]
@baruKreddy : Please don't create new commits while updating PRs. Use `git commit --amend`. Please squash your commits into one.

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/JKubeProjectUtilTest.java [hunk 16]
```diff
@@ -46,16 +54,14 @@ public void hasDependencyWithGroupIdWithNulls() {
   }
 
   @Test
-  public void hasDependencyWithGroupIdWithDependency(@Mocked JavaProject project) {
+  public void hasDependencyWithGroupIdWithDependency() {
     // Given
     // @formatter:off
```
Comment (rohanKanojia) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/JKubeProjectUtilTest.java:59]: Please remove these `// @formatter: off` and `// @formatter: on` statements. These were used earlier to instruct IDE to not autoformat `new Expectations {{` block. Since we've removed it now, it shouldn't be required anymore.

Comment (baruKreddy) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/JKubeProjectUtilTest.java]: Sure,Now removed from all the classes!

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java [hunk 17]
```diff
@@ -38,7 +40,7 @@
 import static org.eclipse.jkube.kit.common.util.EnvUtil.storeTimestamp;
 import static org.junit.Assume.assumeFalse;
 import static org.junit.Assume.assumeTrue;
-
+@RunWith(MockitoJUnitRunner.class)
```
Comment (manusa) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/EnvUtilTest.java:43]: Why is this necessary?

https://github.com/eclipse/jkube/pull/1609#discussion_r897830489

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KubernetesHelperTest.java [hunk 18]
```diff
@@ -53,23 +54,35 @@
...
+@RunWith(MockitoJUnitRunner.class)
 public class KubernetesHelperTest {
 
     private KitLogger logger;
 
+    @Mock
+    private KubernetesClient kubernetesClient;
+
+    @Mock
+    private Resource<Service> svcResource;
```
Comment (manusa) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KubernetesHelperTest.java:84]: Neither the @RunWith(MockitoJUnitRunner.class) annotation
nor these fields are necessary since these variables are used locally and initialized in the test method

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/SpringBootUtilTest.java [hunk 19]
```diff
@@ -33,10 +36,14 @@
...
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
-
+import static org.mockito.Mockito.when;
+@RunWith(MockitoJUnitRunner.class)
 public class SpringBootUtilTest {
     @Rule
     public TemporaryFolder temporaryFolder = new TemporaryFolder();
+    @Mock
+    JavaProject maven_project;
```
Comment (manusa) [inline jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/SpringBootUtilTest.java:45]: Could you please use Java naming conventions for the variable declaration? (I know the original variable was like this, but we definitely need to change this)
```suggestion
    JavaProject mavenProject;
```

File: jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KubernetesHelperTest.java (hunk 18 above)
Comment (baruKreddy) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KubernetesHelperTest.java:84]: @manusa I suppose we cant give parameters inside a test method , so I have removed those variables and declared above globally. If we remove@RunWith(MockitoJUnitRunner.class) in this class test cases are failing.

Comment (baruKreddy) [reply jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/KubernetesHelperTest.java]: Please verify KubernetesHelperTest now! @manusa

**Review by manusa — APPROVED** [review]
